substrate/frame/revive/src/wasm/runtime.rs

```diff
@@ -606,6 +606,29 @@ fn extract_hi_lo(reg: u64) -> (u32, u32) {
 	((reg >> 32) as u32, reg as u32)
 }
 
+/// Provides storage variants to support standard and Etheruem compatible semantics.
+enum StorageValue {
+	/// Indicates that the storage value should be read from a memory buffer.
+	/// - `ptr`: A pointer to the start of the data in sandbox memory.
+	/// - `len`: The length (in bytes) of the data.
+	Memory { ptr: u32, len: u32 },
+
+	/// Indicates that the storage value is provided inline as a fixed-size (256-bit) value.
+	/// This is used by set_storage_or_clear() to avoid double reads.
+	/// This variant is used to implement Ethereum SSTORE-like semantics.
+	Value(Vec<u8>),
+}
+
+/// Controls the output behavior for storage reads, both when a key is found and when it is not.
+enum StorageReadMode {
+	/// Standard mode: if the key exists, the full stored value is returned
+	/// using the caller‑provided output length.
+	Standard { output_len_ptr: u32 },
+	/// Ethereum commpatible mode: always write a 32-byte value into the output buffer.
+	/// If the key is missing, write 32 bytes of zeros.
+	EthSemantics,
```

Comment:
Standard / EthSemantics sounds a bit vague and does not tell you immediately what it is.
maybe something like this 
```rust
enum StorageReadMode {
	VariableOutput { output_len_ptr: u32 }
	FixedOutput32,
}
```